src/routes/safe/components/Transactions/helpers/EditTxParametersForm/index.tsx

```diff
@@ -115,6 +116,22 @@ export const EditTxParametersForm = ({
     onClose()
   }
 
+  const handleSafeNonceChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setHasChangedSafeNonce(true)
+    setManualSafeNonce(e.target.value)
```

Comment:
Can't we just derive `hasChangedSafeNonce` from `manualSafeNonce !== ''`?

Reply:
I tried that but it has slightly different behaviour for when the user removes everything in the input field it will jump back to whatever value `safeNonce` has instead of leaving the input field empty. With the flag it is flipped only once.

Reply:
I usually use `undefined` as initial value in this pattern. You can then tell if the value has been changed even if it's an empty string.

Reply:
Works like a charm 💯